front/src/modules/timesStops/hooks/useTimesStopsTableData.ts

```diff
@@ -0,0 +1,248 @@
+import type { SimulationSummary } from 'modules/timetableItem/types';
+import type { Train } from 'reducers/osrdconf/types';
+import { getDisplayOnlyPathSteps } from 'reducers/simulationResults/selectors';
+import { Duration } from 'utils/duration';
+
+import { getTrackReferenceLabel, getOperationalPointName } from '../helpers/utils';
+import type { ScheduleEntry, TimesStopsRowNew } from '../types';
+
+type BuildTableRowParams = {
+  id: string;
+  name?: string;
+  ch?: string;
+  trackName?: string;
+  startDatetime: Date;
+  schedule?: ScheduleEntry;
+  computedArrivalMs?: number;
+};
+
+const buildTableRow = ({
+  id,
+  name,
+  ch,
+  trackName,
+  startDatetime,
+  schedule,
+  computedArrivalMs,
+}: BuildTableRowParams): TimesStopsRowNew => {
+  // schedule.arrival is ISO 8601 duration
+  const requestedArrivalDuration = schedule?.arrival ? Duration.parse(schedule.arrival) : null;
+  const requestedArrival = requestedArrivalDuration
+    ? new Date(startDatetime.getTime() + requestedArrivalDuration.ms)
+    : null;
+
+  // computedArrivalMs is offset in ms from startDatetime
+  const computedArrival =
+    computedArrivalMs !== undefined ? new Date(startDatetime.getTime() + computedArrivalMs) : null;
+
+  // schedule.stop_for is ISO 8601 duration
+  const stopDuration = schedule?.stop_for ? Duration.parse(schedule.stop_for) : null;
+
+  // requestedDeparture = requestedArrival + stopDuration
+  const requestedDeparture =
+    requestedArrival && stopDuration
+      ? new Date(requestedArrival.getTime() + stopDuration.ms)
+      : null;
+
+  // computedDeparture = computedArrival + stopDuration
+  const computedDeparture =
+    computedArrival && stopDuration ? new Date(computedArrival.getTime() + stopDuration.ms) : null;
+
+  return {
+    id,
+    name: name ?? '',
+    ch: ch ?? '',
+    track: trackName ?? '',
+    requestedArrival,
+    computedArrival,
+    stopDuration,
+    requestedDeparture,
+    computedDeparture,
+  };
+};
+
+/**
+ * Hook to build TableRow[] for the new Times & Stops table (TimesStopsTableNew).
+ * This is separate from useOutputTableData to keep the legacy and new logic independent.
+ * Once the new table is fully validated, useOutputTableData can be removed.
+ */
+const useTimesStopsTableData = (
+  infraId: number,
+  isValid: boolean,
+  selectedTrain?: Train,
+  simulatedTrain?: SimulationResponseSuccess['final_output'],
+  simulatedPathItemTimes?: Extract<SimulationSummary, { isValid: true }>['pathItemTimes'],
+  operationalPointsOnPath?: PathPropertiesFormatted['operationalPoints']
+): TimesStopsRowNew[] => {
+  const { t } = useTranslation('operational-studies');
+  const { getTrackSectionsByIds } = useScenarioContext();
+  const displayOnlyPathSteps = useSelector(getDisplayOnlyPathSteps);
+
+  const pathStepOps = usePathOps(infraId, selectedTrain?.path);
+
+  const trackIds = useMemo(() => {
+    const path = selectedTrain?.path || [];
```

Comment:
You can even rename `path` -> `pathSteps`
```suggestion
    const pathSteps = selectedTrain?.path || [];
```